plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridTable.java

```diff
@@ -0,0 +1,178 @@
+import org.jkiss.dbeaver.ext.cubrid.CubridConstants;
+import org.jkiss.dbeaver.ext.generic.model.GenericTable;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableIndex;
+import org.jkiss.dbeaver.model.DBPEvaluationContext;
+import org.jkiss.dbeaver.model.DBPScriptObject;
+import org.jkiss.dbeaver.model.DBUtils;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.meta.IPropertyValueListProvider;
+import org.jkiss.dbeaver.model.meta.Property;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.DBSObject;
+import org.jkiss.dbeaver.model.struct.DBStructUtils;
+import org.jkiss.utils.CommonUtils;
+
+public class CubridTable extends GenericTable {
+
+	private static final Log log = Log.getLog(CubridTable.class);
+	private String ddl;
+	private CubridUser owner;
+	private CubridObjectContainer container;
+	private boolean isSystemTable;
+
+	public CubridTable(CubridObjectContainer container, @Nullable String tableName, @Nullable String tableType,
+			@Nullable JDBCResultSet dbResult) {
+		super(container, tableName, tableType, dbResult);
+		this.container = container;
+		String owner_name;
+
+		if (dbResult != null) {
+			owner_name = JDBCUtils.safeGetString(dbResult, CubridConstants.OWNER_NAME);
+			isSystemTable = JDBCUtils.safeGetString(dbResult, CubridConstants.IS_SYSTEM_CLASS).equals("YES");
+		} else {
+			owner_name = getDataSource().getContainer().getConnectionConfiguration().getUserName().toUpperCase();
+			isSystemTable = false;
+		}
+
+		for (CubridUser cbOwner : getUsers()) {
+			if (cbOwner.getName().equals(owner_name)) {
+				this.owner = cbOwner;
+				break;
+			}
+		}
+	}
+
+	@Override
+	public boolean isView() {
+		return false;
+	}
+
+	@Override
+	public String getDDL() {
+		return ddl;
+	}
+
+	@Override
+	@Property(hidden = true, editable = true, updatable = true, order = -1)
+	public String getObjectDefinitionText(DBRProgressMonitor monitor, Map<String, Object> options) throws DBException {
+		if (CommonUtils.getOption(options, DBPScriptObject.OPTION_REFRESH)) {
+			ddl = null;
+		}
+		if (!isPersisted()) {
+			return DBStructUtils.generateTableDDL(monitor, this, options, false);
+		}
+		if (ddl == null || !isCacheDDL()) {
+			ddl = getDataSource().getMetaModel().getTableDDL(monitor, this, options);
+		}
+		return ddl;
+	}
+
+	protected boolean isCacheDDL() {
+		return true;
+	}
+
+	@Override
+	public boolean supportsObjectDefinitionOption(String option) {
+		return true;
+	}
+
+	boolean isHideVirtualModel() {
+		return true;
+	}
+
+	public boolean isPhysicalTable() {
+		return !isView();
+	}
+
+	@Override
+	public boolean isSystem() {
+		return isSystemTable;
+	}
+
+	public CubridObjectContainer getContainer() {
+		return this.container;
+	}
+
+	public Collection<? extends CubridUser> getUsers() {
+		try {
+			return container.getDataSource().getCubridUsers(null);

```

Comment:
Avoid using null in monitor parameters better to use
`new VoidProgressMonitor()`, but I would recommend reading my comments about passing the session monitor to properly handle loading any metadata

Reply:
I have fixed it.